src/app/modules/main/chat_section/module.nim

```diff
@@ -952,21 +952,33 @@ method changeMutedOnChat*(self: Module, chatId: string, muted: bool) =
 proc changeCanPostValues*(self: Module, chatId: string, canPost, canView, canPostReactions, viewersCanPostReactions: bool) =
   self.view.chatsModel().changeCanPostValues(chatId, canPost, canView, canPostReactions, viewersCanPostReactions)
 
+proc updateChatsRequiredPermissions(self: Module, chats: seq[ChatDto]) =
+  for chat in chats:
+    self.updateChatRequiredPermissions(chat.id)
+
+proc displayTokenPermissionChangeNotification(self: Module, title: string, message: string, community: CommunityDto, tokenPermission: CommunityTokenPermissionDto) =
+  if self.showPermissionUpdateNotification(community, tokenPermission):
+    singletonInstance.globalEvents.showCommunityTokenPermissionCreatedNotification(community.id, title, message)
+
 method onCommunityTokenPermissionDeleted*(self: Module, communityId: string, tokenPermission: CommunityTokenPermissionDto) =
+  self.view.tokenPermissionsModel.removeItemWithId(tokenPermission.id)
+
   self.rebuildCommunityTokenPermissionsModel()
```

Comment:
Do you think it does make sense to rebuild the model if you already removed the item?

Reply:
Sure, makes sense to just call `reevaluateRequiresTokenPermissionToJoin()` for consistency. Not necessary to rebuild the model from there 👍🏾